packages/pfsense/data_stream/log/elasticsearch/ingest_pipeline/haproxy.yml

```
@@ -105,21 +73,14 @@ processors:
       type: long
       ignore_missing: true
       if: ctx.containsKey('http')
-  - set:
-      field: event.kind
-      value: event
   - append:
       field: event.category
       value: web
       if: "ctx?.haproxy?.mode == 'HTTP' || ctx?.haproxy?.http != null"
-  - append:
-      field: event.category
-      value: network
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
   - append:
       field: event.type
       value: connection
-      if: "ctx?.source.ip != null && ctx?.destination?.ip != null"
+      if: "ctx?.source.address != null && ctx?.destination?.address != null"
```

```suggestion
      if: "ctx.source.address != null && ctx.destination?.address != null"
```

Also query that we know `ctx.source` is non-null at this point? Should a null-safe selector be added to that access?